NonsoAmadi10/Auto-Mart
# #166699738 Creates User signup feature

### What does this PR do?
- Creates ` POST: /api/auth/signup/ `

### Description of Task taken to complete??
- setup directories
- implement feature with a database
- call database queries
- expect a token in the response

### How should this be manually Tested?
- Clone this repo
- ` git checkout ft-user-should-be-able-to-signup-166699738`
- Run ` npm install `
- start the server ` npm run dev-start `
- Open postman on  port ` 4200` , then ` POST localhost:4200 /api/v1/auth/signup `

### Any Background Context you want to provide?
- the request body must contain the following entities:
 ```
{
"firstname": String,
"lastname": String,
"email": String,
"password" : String,
"confirmPassword": String,
"address": String,
"adminSecret": String
} 

```
- You must setup all .env configs in the .env.example file
- To signup as an admin, your adminSecret must be equal to your process.env.ADMIN_SECRET
- To see test cases ` Run npm test ` in the command line


### What are the relevant pivotal tracker stories??
[#166699738](https://www.pivotaltracker.com/story/show/166699738)
